theory/theory.md

```diff
@@ -0,0 +1,362 @@
+
+Here is an example of polymorphism with Classes: 
+
+    class Adult:
+        def move(self):
+            print('walk')
+    
+        def greet(self):
+            print('Hello!')
+
+
+    class Baby:
+        def move(self):
+            print('crawl')
+    
+        def greet(self):
+            print('smile :)')
+
+
+    Keiko = Adult()
+    Emma = Baby()
+    
+    for human in (Keiko, Emma):
+        human.move()
+        human.greet()
+    
+    # output
+    # walk
+    # Hello!
+    # crawl
+    # smile :)
+
+Here, we have created two classes Adultand Baby. They share a similar structure and have the same method names move() and greet(). Although we have not created a common superclass or linked the classes together, we can pack these two different objects into a tuple and iterate through it using a common human variable. It is possible due to polymorphism. 
+
+
+Source: https://www.programiz.com/python-programming/polymorphism
+
+## 3. What's inheritance in OOP?
+
+The ability to define a new class that is a modified version of an existing class is known as inheritance.
+
+This feature's main benefit is that you can add new methods to classes without changing the ones that already exist. Because the new class inherits all of the existing class's methods, it is called "inheritance." Using this metaphor further, the current class may also be referred to as the parent class. The new class may be referred to as a "subclass" or its "child class."
+
+Inheritance is a useful feature because inheritance can also help with code reuse without having to modify them. In some circumstances, the inheritance structure reflects the problem's natural nature, making the programme easier to understand.
+
+Inheritance, on the other hand, can make programmes harder to read. When a method is called, it is not always obvious where to look for its definition. The relevant code may be spread over numerous modules. Furthermore, many of the things that may be done using inheritance may be done just as neatly without inheritance. 
+
+Following is an example of inheritance where a class named Car inherited properties and methods from a class named Vehicle.
+
+Create a class named Vehicle, with brand and name properties, and a printname method:
+
+    class Vehicle:
+      def __init__(self, brand, name):
+        self.brand = brand
+        self.name = name
+    
+      def printname(self):
+        print(self.brand, self.name)
+    
+    x = Vehicle("Toyota", "Sora Bus")
+    x.printname()
+
+    # Create a class named Car, which will inherit the properties and methods from the Vehicle class:
+    class Car(Vehicle):
+      pass
+    
+    # Now the Car class has the same properties and methods as the Vehicle class.
+    x = Car("Ford", "F-Series")
+    x.printname()
+
+Source : https://openbookproject.net/thinkcs/python/english3e/inheritance.html#:~:text=Inheritance%20is%20the%20ability%20to,methods%20of%20the%20existing%20class.
+

```

Comment:
9 / 10
Lovely attempt, Keiko. One thing i was looking out for, but could not find was a place you must've mentioned the obvious concept about hierarchy in inheritance.